tests/test_masked_autoencoder_vit.py

```diff
@@ -0,0 +1,260 @@
+                dropout_rate=0.3,
+            )
+
+        with self.assertRaises(ValueError):
+            MaskedAutoEncoderViT(
+                in_channels=1,
+                img_size=(32, 32, 32),
+                patch_size=(64, 64, 64),
+                hidden_size=512,
+                mlp_dim=3072,
+                num_layers=12,
+                num_heads=8,
+                decoder_pos_embed_type="sin",
+                dropout_rate=0.3,
+            )
+        with self.assertRaises(ValueError):
+            MaskedAutoEncoderViT(
+                in_channels=1,
+                img_size=(32, 32, 32),
+                patch_size=(64, 64, 64),
+                hidden_size=512,
+                mlp_dim=3072,
+                num_layers=12,
+                num_heads=8,
+                proj_type="perceptron",
+                dropout_rate=0.3,
+            )
+
+        with self.assertRaises(ValueError):
+            MaskedAutoEncoderViT(
+                in_channels=1,
+                img_size=(96, 96, 96),
+                patch_size=(8, 8, 8),
+                hidden_size=512,
+                mlp_dim=3072,
+                num_layers=12,
+                num_heads=14,
+                proj_type="conv",
+                dropout_rate=0.3,
+            )
+
+        with self.assertRaises(ValueError):
+            MaskedAutoEncoderViT(
+                in_channels=1,
+                img_size=(97, 97, 97),
+                patch_size=(4, 4, 4),
+                hidden_size=768,
+                mlp_dim=3072,
+                num_layers=12,
+                num_heads=8,
+                proj_type="perceptron",
+                dropout_rate=0.3,
+            )
+
+        with self.assertRaises(ValueError):
+            MaskedAutoEncoderViT(
+                in_channels=4,
+                img_size=(96, 96, 96),
+                patch_size=(16, 16, 16),
+                hidden_size=768,
+                mlp_dim=3072,
+                num_layers=12,
+                num_heads=12,
+                proj_type="perc",
+                dropout_rate=0.3,
+            )
+        with self.assertRaises(ValueError):
+            MaskedAutoEncoderViT(
+                in_channels=4,
+                img_size=(96, 96, 96),
+                patch_size=(16, 16, 16),
+                hidden_size=768,
+                mlp_dim=3072,
+                num_layers=12,
+                num_heads=12,
+                proj_type="conv",
+                dropout_rate=0.3,
+                masking_ratio=1.1,
+            )
+        with self.assertRaises(ValueError):
+            MaskedAutoEncoderViT(
+                in_channels=4,
+                img_size=(96, 96, 96),
+                patch_size=(16, 16, 16),
+                hidden_size=768,
+                mlp_dim=3072,
+                num_layers=12,
+                num_heads=12,
+                proj_type="conv",
+                dropout_rate=0.3,
+                masking_ratio=-0.1,
+            )
```

Comment:
This could be implemented by putting the arguments of each construction into a list and using parameterized.expand.

Reply:
Yes, thanks for the comment, I've corrected it now in the new push!